src/modules/0.0.22/aws_ecr/rpcs/build.ts

```diff
@@ -0,0 +1,167 @@
+        let credentialsArn;
+        if (githubPersonalAccessToken) {
+            const createCredentialsInput: ImportSourceCredentialsInput = {
+                token: githubPersonalAccessToken,
+                serverType: 'GITHUB',
+                authType: 'PERSONAL_ACCESS_TOKEN',
+            };
+            credentialsArn = await this.importSourceCredentials(client.cbClient, createCredentialsInput);
+            if (!credentialsArn) throw new Error('Error adding Github credentials');
+        }
+
+        // create service role
+        new IamRole()
+        awsIamModule.role.cloud.create()
+
+        // create codebuild project
+        const ecrRepositoryUri = (await this.module.repository.db.read(ctx) as Repository[]).find(repo => repo.id === parseInt(ecrRepositoryId))?.repositoryUri;
+        const buildspec = `version: 0.2
+        
+        phases:
+          pre_build:
+            commands:
+              - echo Logging in to Amazon ECR...
+              - aws ecr get-login-password --region ${region ?? ctx.getDefaultRegion()} | docker login --username AWS --password-stdin ${ecrRepositoryUri}
+          build:
+            commands:
+              - echo Building the Docker image...
+              - docker build -t ${ecrRepositoryUri}:latest .
+          post_build:
+            commands:
+              - echo Pushing the Docker image...
+              - docker push ${ecrRepositoryUri}:latest`;
+        const prefix = (Math.random() + 1).toString(36).substring(7);;
+        const codeBuildProjectName = `${prefix}-ecr-builder`;
+        const createProjectInput: CreateProjectCommandInput = {
+            name: codeBuildProjectName,
+            environment: {
+                type: 'LINUX_CONTAINER',
+                image: 'aws/codebuild/standard:6.0',
+                computeType: 'BUILD_GENERAL1_SMALL',
+                environmentVariables: [],
+                privilegedMode: true,
+            },
+            sourceVersion: 'main',
+            source: {
+                location: githubRepoUrl,
+                type: 'GITHUB',
+                buildspec: buildspec,
+            },
+            serviceRole: undefined,
+            artifacts: undefined
+        };
+        const awsPj = await this.createProject(client.cbClient, createProjectInput);
+        if (!awsPj) throw new Error("Couldn't create CodeBuild project");
+
+        // start codebuild project
+        const startProjectInput: StartBuildInput = {
+            projectName: codeBuildProjectName,
+        };
+        const cloudBuild = await this.startBuild(client.cbClient, startProjectInput);
+        if (!cloudBuild || !cloudBuild.id) throw new Error('Error starting build');
+        await this.waitForBuildsToComplete(client.cbClient, [cloudBuild.id]);
+
+        const imageId = (await this.module.repositoryImages.cloud.read(ctx) as RepositoryImage[]).find(repo => repo.privateRepository?.id === parseInt(ecrRepositoryId))?.imageId;
+
```

Comment:
I think we need to do a db create for the ecr image somewhere here?

Reply:
yeah here we should be calling the ecr images `db.create` method